### PR TITLE
feat(suite): implement the coinjoin summary status component

### LINKED ISSUE #6219: CoinJoin progress and it's controls
CJ - progress bar

On hover to three dots show:
pause or cancel cj

Show blocking modal if user wants to cancel the CoinJoin
Cancel Plaholder

### PR DESCRIPTION
**Pull request by dahaca**
## Description

Implemented the Coinjoin status wheel UI. Like with the [status bar](https://github.com/trezor/trezor-suite/pull/6370), the UI and functionality are not final, once the coinjoin API (phases, estimates calculateion, pausing) is finalised, additional work has to be done.

## Related Issue

Resolve [#6219](https://github.com/trezor/trezor-suite/issues/6219)

## Screenshots:

image

### CODE REVIEW

File: packages/suite/src/support/messages.ts [hunk 1]
```diff
@@ -7054,4 +7054,49 @@ export default defineMessages({
...
+        id: 'TR_CANCEL_COINJOIN_YES',
+        defaultMessage: 'Yes, cancel it',
+    },
+    TR_PAUSED: {
+        id: 'TR_PAUSED',
+        defaultMessage: 'Paused',
+    },
+    TR_RESUME: {
+        id: 'TR_RESUME',
+        defaultMessage: 'Pesume',
```
Comment (matejkriz) [inline packages/suite/src/support/messages.ts]: typo
```suggestion
        defaultMessage: 'Resume',
```

File: packages/suite/src/support/messages.ts [hunk 2]
```diff
@@ -7054,4 +7054,49 @@ export default defineMessages({
         description: 'Button in coin join settings',
         defaultMessage: 'Reset to default',
     },
+    TR_LEFT: {
+        id: 'TR_LEFT',
+        description: 'As in "{time} Left"',
+        defaultMessage: 'Left',
```
Comment (matejkriz) [inline packages/suite/src/support/messages.ts]: nit:
```suggestion
        defaultMessage: 'left',
```

File: packages/suite/src/components/wallet/CoinjoinSummary/BalanceSection.tsx [hunk 3]
```diff
@@ -2,12 +2,12 @@ import React from 'react';
 import styled from 'styled-components';
 
 import * as routerActions from '@suite-actions/routerActions';
-import { Account } from '@suite-common/wallet-types';
+import { Account, CoinjoinSession } from '@suite-common/wallet-types';
 import { Card, Translation } from '@suite-components';
 import { useActions, useSelector } from '@suite-hooks';
 import { Button } from '@trezor/components';
-import { CoinjoinSessionStatus } from './CoinjoinSessionStatus';
```
Comment (matejkriz) [inline packages/suite/src/components/wallet/CoinjoinSummary/BalanceSection.tsx:9]: You could also delete the file, right? It is replaced by CoinjoinStatus...

File: packages/suite/src/components/wallet/CoinjoinSummary/BalanceSection.tsx [hunk 4]
```diff
@@ -39,29 +51,21 @@ export const BalanceSection = ({ account }: BalanceSectionProps) => {
...
 
     return (
-        <>
-            {/* temporary - content will be reworked and moved into the container below */}
-            {session && (
-                <Container>
-                    <CoinjoinSessionStatus account={account} session={session} />
-                </Container>
-            )}
+        <Container>
```
Comment (matejkriz) [inline packages/suite/src/components/wallet/CoinjoinSummary/BalanceSection.tsx:54]: not responsive enough
image

**Comment by matejkriz** [comment]
Is the global cross icon change intentional? I personally liked the previous proportions better, but that is just my taste.

File: packages/suite/src/components/wallet/CoinjoinSummary/CoinjoinStatus.tsx [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    const iconConfig = {
+        size: 18,
+        color: theme.TYPE_DARK_GREY,
+    };
+
+    return (
+        <Container>
+            <SessionControlsMenu alignMenu="right" items={menuItems} ref={menuRef} />
+
+            <ProgressWheel
```
Comment (matejkriz) [inline packages/suite/src/components/wallet/CoinjoinSummary/CoinjoinStatus.tsx:195]: Very nice! 👍 
image
image

...

**Review by matejkriz — CHANGES_REQUESTED** [review]

**Comment by dahaca** [comment]
> Is the global cross icon change intentional? I personally liked the previous proportions better, but that is just my taste.

The one on the image in the description is not correct, I made is smaller. :) Previously, it had too much space around and I had to treat it differently not only in this PR.

It is too different now somewhere?

File: packages/suite/src/components/wallet/CoinjoinSummary/BalanceSection.tsx (hunk 3 above)
Comment (dahaca) [reply packages/suite/src/components/wallet/CoinjoinSummary/BalanceSection.tsx:9]: https://github.com/trezor/trezor-suite/pull/6409/commits/9911012d094a8b339d2a4eda849e085d80041d6f

File: packages/suite/src/support/messages.ts (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/support/messages.ts]: https://github.com/trezor/trezor-suite/pull/6409/commits/9911012d094a8b339d2a4eda849e085d80041d6f

File: packages/suite/src/support/messages.ts (hunk 1 above)
Comment (dahaca) [reply packages/suite/src/support/messages.ts]: https://github.com/trezor/trezor-suite/pull/6409/commits/9911012d094a8b339d2a4eda849e085d80041d6f

**Comment by matejkriz** [comment]
> 



> > Is the global cross icon change intentional? I personally liked the previous proportions better, but that is just my taste.
> 
> The one on the image in the description is not correct, I made is smaller. :) Previously, it had too much space around and I had to treat it differently not only in this PR.
> 
> It is too different now somewhere?

Just go and check the other occurrences.

e.g. 
Settings Close button
image
image

close buttons in some modals
image
image

and so on
image
image


I didn't come across anything broken, I was just wondering if the intention was to change it this globally.

**Review by matejkriz — APPROVED** [review]

**Comment by dahaca** [comment]
> I was just wondering if the intention was to change it this globally.

Not at all, reverted back, it was broken in too many places.

File: packages/suite/src/components/wallet/CoinjoinSummary/BalanceSection.tsx (hunk 4 above)
Comment (dahaca) [reply packages/suite/src/components/wallet/CoinjoinSummary/BalanceSection.tsx:54]: https://github.com/trezor/trezor-suite/issues/6473

**Pull request merged**